📖  Add troubleshooting advice when running Quick Start with CAPD

oscr opened the pull request:
**What this PR does / why we need it**:

The Quick Start with CAPD didn't work on my machine and now I've figured out why. To help others that might run into the same problem I've added some troubleshooting advice to the Quick Start. I also sneaked in some minor corrections.  

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

killianmuldoon commented:
I think this is better for the troubleshooting guide rather than the quick start - we already have some information on this problem in there, but maybe a direct link to the troubleshooting guide (or this issue in particular) would be useful to help people discover this: https://cluster-api.sigs.k8s.io/user/troubleshooting.html#cluster-api-with-docker----too-many-open-files

oscr commented:
Thank you for the advice Killian. I somehow missed the Troubleshooting Guide. I've added references to it from the QS and moved the machine troubleshooting part to it.

killianmuldoon commented:
I really like this info, and I think the structure of it could really help people out.  It doesn't realy fit in with the structure of our Troubleshooting though, which is more like a "Common issues" page.

What do you think about making this a "Step-by-step troubleshooting example" as the first item in the page?

It would be good to expand it a little bit to include info on how you got to that point - e.g. following the quickstart for Docker - and the different steps you followed that brought you to decide the machine was the problem - e.g. stuck in bootstrapping for too long. So it's something a new user would like at to find out the basic process of investigating issues in CAPI

killianmuldoon commented on docs/book/src/user/troubleshooting.md:
````diff
@@ -1,5 +1,64 @@
+    └─3 Machines...                                            True                                          47m    See capi-quickstart-md-0-d6dn6-584ff97cb7-kr7bj, capi-quickstart-md-0-d6dn6-584ff97cb7-s6cbf, ...
+```
+
+Machines should be started, but Workers are not ready due to Calico isn't installed yet. You should be able to see the containers running with `docker ps --all` and they should not be restarting.
````
```suggestion
Machines should be started, but Workers are not because Calico isn't installed yet. You should be able to see the containers running with `docker ps --all` and they should not be restarting.
```

oscr commented on docs/book/src/user/troubleshooting.md:
```diff
@@ -1,5 +1,64 @@
+
+In the example above we can see that the Machine `capi-quickstart-6587k-xtvnz` has failed to start. The reason provided is `BootstrapFailed`. 
+
+To investigate why a machine fails to start you can inspect the logs using `docker logs <MACHINE-NAME>`. For example:
```
Great addition! Thank you. It's been added

killianmuldoon reviewed (approved):
/lgtm

Great work!

oscr commented:
Thank you Killian for your great feedback!

ykakarap reviewed (commented):
This is a great addition and it is well structured!

Great work @oscr 🚀

/lgtm

fabriziopandini commented:
Great work
/lgtm
/approve

killianmuldoon commented:
/cherry-pick release-1.2
